autoprecompiles/src/powdr.rs

```diff
@@ -128,18 +131,24 @@ pub fn substitute_algebraic_algebraic<T: Clone + std::cmp::Ord>(
     );
 }
 
+// @leo: This is where I'm not sure. I think after this change this function should return `Column -> Column`, which it does, but I'm not sure my implementation is correct
```

Comment:
@leonardoalt

Reply:
looks fine I think?

Reply:
the `expr` keeps its own poly id which is important, and the name is changed in place. The return is only for the substitution map returned later on, so looks fine